relayer/relayer.go

```diff
@@ -343,26 +409,33 @@ func (r *Relayer) RelayMessage(warpLogInfo *vmtypes.WarpLogInfo, storeProcessedH
 		)
 		return err
 	}
+	if !r.CheckAllowedDestinationAddress(destinationAddress, destinationBlockchainID) {
+		r.logger.Debug(
+			"Message destination address not allowed. Not relaying.",
+			zap.String("blockchainID", r.sourceBlockchain.GetBlockchainID().String()),
+			zap.String("destinationBlockchainID", destinationBlockchainID.String()),
+			zap.String("destinationAddress", destinationAddress.String()),
+		)
+		return nil
+	}
 
 	// Check that the destination chain ID is supported
 	if !r.CheckSupportedDestination(destinationBlockchainID) {
 		r.logger.Debug(
 			"Message destination chain ID not supported. Not relaying.",
 			zap.String("blockchainID", r.sourceBlockchain.GetBlockchainID().String()),
-			zap.String("sourceBlockchain.GetBlockchainID()", destinationBlockchainID.String()),
+			zap.String("destinationBlockchainID", destinationBlockchainID.String()),
 		)
 		return nil
 	}
 
-	messageRelayerID := database.CalculateRelayerID(
+	messageRelayer, ok := r.getMessageRelayer(
```

Comment:
Is this check sufficient for removing the `CheckSupportedDestination`, `CheckAllowedDestinationAddress` and `CheckAllowedOriginSenderAddress` checks above? I think if it's not a supported source/destination, then we won't have a message relayer for it.

Reply:
It totally is, great call. Removed all of those checks. That should also resolve a handful of your other comments :)